components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/EndpointRegistryImpl.java

```diff
@@ -78,6 +78,21 @@ public EndpointRegistryInfo getEndpointRegistryByUUID(String registryId) throws
         return apiMgtDAO.getEndpointRegistryByUUID(registryId);
     }
 
+    /**
+     * Deletes an Endpoint Registry
+     *
+     * @param registryId Registry Identifier(UUID)
+     * @throws APIManagementException if failed to delete the Endpoint Registry
+     */
+    public void deleteEndpointRegistry(String registryId) throws APIManagementException {
+        EndpointRegistryInfo endpointRegistryInfo = apiMgtDAO.getEndpointRegistryByUUID(registryId);
```

Comment:
Shall we do this in the REST API level, so that we can throw the exception from there itself? I felt the readability is a bit reduced when we throw it from here and catch it from the parent and then handle the exception again.

Reply:
if you agree with this approach, you may change the same in other places as well.

Reply:
Fixed with the commit **dd74437**